### PR TITLE
chore: rename algo to function

### PR DESCRIPTION
**Pull request by ThibaultFy**
## Description

Rename Algo concept to Function on all the stack.

## Companion PRs

- https://github.com/Substra/orchestrator/pull/139
- https://github.com/Substra/substra-backend/pull/573
- https://github.com/Substra/substra-tests/pull/243
- https://github.com/Substra/substra/pull/341
- https://github.com/Substra/substrafl/pull/82
- https://github.com/Substra/substra-documentation/pull/264
- https://github.com/Substra/substra-frontend/pull/170

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,10 +7,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
+### Changed
+
+-   BREAKING: rename Algo to Function (#170)
+
 ## [0.39.2] - 2022-02-09
 
 ### Changed
 
+-   BREAKING: rename Algo to Function (#170)
```
Comment (Milouu) [inline CHANGELOG.md:18]: Need to remove this line 🔥

File: e2e-tests/cypress/integration/functions.spec.js [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+    it('lists functions', () => {
+        cy.visit('/functions');
         cy.get('tbody[data-cy=loaded]')
             .get('tr')
             .should('have.length.greaterThan', 2);
     });
 
-    it('displays an algo drawer', () => {
-        cy.visit('/algorithms');
+    it('displays an function drawer', () => {
```
Comment (Milouu) [inline e2e-tests/cypress/integration/functions.spec.js:11]:
```suggestion
    it('displays a function drawer', () => {
```

File: CHANGELOG.md (hunk 1 above)
Comment (ThibaultFy) [reply CHANGELOG.md]: Yes merci ! done :)

**Review by Milouu — CHANGES_REQUESTED** [review]
Little grammar correction but lgtm to me otherwise. Not tested yet

**Review by Milouu — APPROVED** [review]

**Pull request merged**